cmd/image-builder-db-test/main_test.go

```diff
@@ -71,75 +71,97 @@ func migrateUp(t *testing.T) {
+	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR1, ORGID1)
+	require.Error(t, err)
 
-	insert = "INSERT INTO composes(job_id, request, created_at, account_id, org_id) VALUES ($1, $2, CURRENT_TIMESTAMP - interval '3 days', $3, $4)"
+	migrateOneStep(t) // migrate to step 3
 
-	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR3, ORGID1)
-	require.NoError(t, err)
+	// Verify that after migration step 3 adding a compose request to the db requires a non empty account number.
+	d, err := db.InitDBConnectionPool(connStr(t))
+	err = d.InsertCompose(uuid.New().String(), "", ORGID1, []byte("{}"))
+	require.Error(t, err)
 
-	insert = "INSERT INTO composes(job_id, request, created_at, account_id, org_id) VALUES ($1, $2, CURRENT_TIMESTAMP - interval '4 days', $3, $4)"
+	// make sure migrating a fully migrated db doesn't error out
+	migrateUp(t)
 
-	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR3, ORGID1)
+	// Check data inserted at migration step 1 and 2 are still accessible
+	_, count, err := d.GetComposes(ANR1, 100, 0)
 	require.NoError(t, err)
+	require.Equal(t, 1, count)
+	_, count, err = d.GetComposes(ANR2, 100, 0)
+	require.NoError(t, err)
+	require.Equal(t, 1, count)
 }
 
-func insertDataForMigrationStep2(t *testing.T) {
-	conn := connect(t)
-	defer conn.Close(context.Background())
-
-	insert := "INSERT INTO composes(job_id, request, created_at, account_number, org_id) VALUES ($1, $2, CURRENT_TIMESTAMP, $3, $4)"
-
-	_, err := conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR1, ORGID1)
+func testInsertCompose(t *testing.T) {
+	d, err := db.InitDBConnectionPool(connStr(t))
 	require.NoError(t, err)
 
-	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR1, ORGID1)
-	require.NoError(t, err)
+	// teardwon
+	defer migrateDown(t)
 
-	_, err = conn.Exec(context.Background(), insert, uuid.New().String(), "{}", ANR2, ORGID1)
+	// setup
+	migrateUp(t)
+
+	// test
+	err = d.InsertCompose(uuid.New().String(), ANR1, ORGID1, []byte("{}"))
 	require.NoError(t, err)
+	err = d.InsertCompose("toto", ANR1, ORGID1, []byte("{}"))
+	require.Error(t, err)
+	err = d.InsertCompose(uuid.New().String(), "", ORGID1, []byte("{}"))
+	require.Error(t, err)
 }
 
-func TestMain(t *testing.T) {
-	migrateOneStep(t)
-	insertDataForMigrationStep1(t)
-
-	migrateOneStep(t)
-	insertDataForMigrationStep2(t)
+func testGetCompose(t *testing.T) {
+	d, err := db.InitDBConnectionPool(connStr(t))
+	require.NoError(t, err)
 
-	migrateOneStep(t)
+	// teardwon
+	defer migrateDown(t)
 
-	// make sure migrating a fully migrated db doesn't error out
+	// setup
 	migrateUp(t)
+	err = d.InsertCompose(uuid.New().String(), ANR1, ORGID1, []byte("{}"))
+	err = d.InsertCompose(uuid.New().String(), ANR1, ORGID1, []byte("{}"))
+	err = d.InsertCompose(uuid.New().String(), ANR1, ORGID1, []byte("{}"))
+	err = d.InsertCompose(uuid.New().String(), ANR1, ORGID1, []byte("{}"))
```

Comment:
Check the errors here.

Reply:
I'm not sure we should, since it'll blow up in the test just below.

Reply:
failing earlier is better, it'll make it easier to debug. Failures here are unlikely but you'd get an exact line number etc...

Reply:
Good point indeed.